conductor/src/client_state.cpp

```diff
@@ -0,0 +1,157 @@
+
+#include "client_state.hpp"
+
+namespace cocaine { namespace isolate { namespace conductor {
+
+namespace state {
+
+
+base_t::base_t(shared_ptr<client_t> parent):
+    m_parent(parent)
+{
+    // Empty
+}
+
+
+
+void
+closed_t::enqueue(shared_ptr<action::action_t> action){
+    m_parent->inbox.push_back(action);
+    connect();
+}
+
+void
+closed_t::cancel(uint64_t request_id){
+    // remove action from inbox
+    // NOP, since the action will be filtered later in parent->enqueue()
+}
+    
+void
+closed_t::connect(){
+    if (m_parent->state.get() == this){
+        m_parent->state = make_shared<connecting_t>(m_parent);
+    }
+}
+
+
+connecting_t::connecting_t (shared_ptr<client_t> parent):
+    m_parent(parent)
+{
+    reschedule_connect();
+}
+
+void
+connecting_t::enqueue(shared_ptr<action::action_t> action){
+    m_parent->inbox.push_back(action);
```

Comment:
Can't we consider states as implementation details of client_t? What would be the idiomatic way to do that?